contracts/Faucet.sol

```diff
@@ -9,113 +9,105 @@ import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
 contract Faucet is Ownable {
     using SafeERC20 for IERC20;
 
+    uint256 public constant TIMEOUT_LIMIT = 30;
+    uint256 public constant MAX_WITHDRAWAL_AMOUNT = 5 ether;
     address public contractRegistry;
-    address public bctAddress;
-    address public nctAddress;
-    mapping(address => uint256) private lastWithdrawalTimes;
+    mapping(address => bool) public isPoolEligible;
+    mapping(address => uint256) private _lastWithdrawalTimes;
     event Deposited(address erc20Addr, uint256 amount);
     event Withdrawn(address account, address erc20Addr, uint256 amount);
 
-    constructor(
-        address _contractRegistry,
-        address _bctAddress,
-        address _nctAddress
-    ) {
-        contractRegistry = _contractRegistry;
-        bctAddress = _bctAddress;
-        nctAddress = _nctAddress;
+    constructor(address contractRegistry_, address[] memory poolAddresses) {
+        contractRegistry = contractRegistry_;
+
+        uint256 poolAddressesCount = poolAddresses.length;
+        for (uint256 i = 0; i < poolAddressesCount; i++) {
+            setPoolEligible(poolAddresses[i], true);
+        }
     }
 
     /// @notice change the TCO2 contracts registry
-    /// @param _address the new contract registry address
+    /// @param contractRegistry_ the new contract registry address
     function setToucanContractRegistry(
```

Comment:
Another thing to think about is registering TCO2 factories instead of the contract registry. Then in a similar fashion to [this comment](https://github.com/ToucanProtocol/faucet/pull/8/files#r1494864237), the faucet could expose all TCO2s that a user holds or can withdraw by calling `getContracts()` in all factories.